util/file.go

```diff
@@ -253,20 +264,110 @@ func JoinTerraformModulePath(modulesFolder string, path string) string {
 	return fmt.Sprintf("%s//%s", cleanModulesFolder, cleanPath)
 }
 
-// Returns true if the given path contains the given folder name.
-//
-// Examples:
-//
-// PathContains("/foo/bar", "foo") => returns true
-// PathContains("/foo/bar", "baz") => returns false
-func PathContains(path string, folderName string) bool {
-	pathParts := strings.Split(path, string(filepath.Separator))
+// fileManifest represents a manifest with paths of all files copied by terragrunt.
+// This allows to clean those files on subsequent runs.
+// The problem is as follows: terragrunt copies the terraform source code first to "working directory" using go-getter,
+// and then copies all files from the directory it runs from to the above dir.
+// It works fine on the first run, but if we delete a file from the current terragrunt directory, we want it
+// to be cleaned in the "working directory" as well. Since we don't really know what can get copied by go-getter,
+// we have to track all the files we tough in a manifest. This way we know exactly which files we need to clean on
+// subsequent runs.
+type fileManifest struct {
+	Path         string
+	ManifestFile string
```

Comment:
What is `Path` vs `ManifestFile`? Is the former the folder path and the latter the path to the manifest file within that folder?

Reply:
Renamed `Path` to `ManifestFolder` and added comments to each var. It used to be just `Path` before handling the subdirectories properly.